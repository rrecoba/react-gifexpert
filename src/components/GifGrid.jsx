import { useState, useEffect } from 'react'
//import { getGifs } from '../helpers/getGifs';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifItem } from './GifItem';

export const GifGrid = ({ category }) => {

    //Custom Hook
    const { images, isLoading } = useFetchGifs ( category ) ;

    return (
        <>
            <h1>{ category }</h1>
            {
                isLoading ? (  <h2>Cargando...</h2>) : null
            }
           

            <div className='card-grid'>
            { 
                images.map( ( img ) => (
                    <GifItem 
                        key={ img.id }
                        title={ img.title } 
                        url={ img. url }
                    />
                ))
            }
            </div>
        </>
    )
}
