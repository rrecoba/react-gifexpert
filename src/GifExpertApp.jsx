import React, { useState } from 'react'
import { AddCategory, GifGrid } from './components/index';


export const GifExpertApp = () => {
  
    const [categories, setCategories] = useState(['One Punch']);
    
    const onAddCategory = (newCategory) => {
        if (categories.includes(newCategory)) return;
        setCategories([...categories, newCategory]);
    }

    return (
    <>   
        {/* titulo */ }
        <h1>GifExpertApp</h1>

        {/* Input */ }
        <AddCategory onNewCategory={ (newCat)=> onAddCategory(newCat) } />

        {/* Listado de Gif */ }
        { 
            categories.map( ( category ) => (
                <GifGrid 
                    key={ category } 
                    category={ category }
                />
            ))
        }
    </>
  )
}
