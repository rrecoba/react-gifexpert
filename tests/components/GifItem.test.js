import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components"

describe('Puebas en <GifItem/>', () => {

    const title = 'pepe';
    const url = 'http://google.com/'

    test('debe de hacer match con el snapshot', () => { 

        const { container } = render(<GifItem title={ title } url={ url }/>);
        expect( container ).toMatchSnapshot();
    })    

    test('debe de mostrar la imagen con el url y alt indicando', () => { 
        
        render( <GifItem title={ title } url={ url }/> );
        //screen.debug();
        const { src, alt } = screen.getByRole('img')
        expect( src ).toBe( url );
        expect( alt ).toBe( title );
    })

     test('debe mostrar el titulo', () => { 

        render( <GifItem title={ title } url={ url }/> );
        expect( screen.getByText( title) ).toBeTruthy();
    })
})