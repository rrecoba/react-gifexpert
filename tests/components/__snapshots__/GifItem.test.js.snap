// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Puebas en <GifItem/> debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="pepe"
      src="http://google.com/"
    />
    <p>
      pepe
    </p>
  </div>
</div>
`;
